sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -2145,14 +2204,21 @@ private async Task RunProducerManagementAsync(CancellationToken cancellationToke
                                 // Responsibility for releasing the guard semaphore is passed to the task.
 
                                 activeTasks.Add(PublishBatchToPartition(partitionState, releaseGuard: true, activeOperationCancellationSource.Token));
+                                consecutiveEmptyBufferCount = 0;
```

Comment:
perhaps I'm missing something, but where do we loop and allow this value to increment past 1?

Reply:
It's on L2217, each time the delay calculates it increments.

Reply:
I see that increment from 0 to 1, but what is the path to get to 2? From what I can see the `while` loop is above this on L2157, so it seems like we need to execute this line each iteration.

Reply:
Don't we?   Each tick of that loop, if we have no active publishing tasks, we'll hit this block and evaluate if we have nothing active because there are no events or because we haven't gotten to their partition yet.    When the answer is "no events", we'll continue to hit the delay calculate and increment.    

That gets reset when we start a publishing task, and we'll short-circuit if there are events waiting but we haven't hit that partition yet.

Am I overlooking something?

Reply:
I might be a little confused- but maybe he's saying that it looks like the value gets reset to 0 every iteration? Like it should only be set to 0 when there are active tasks again?

Reply:
@m-redding that is exactly what I was saying :)

Reply:
chatted with @jsquire offline - I was missing that the if block on 2199 prevents the reset when the buffer is still empty

Reply:
Chris and I discussed offline; the logic is correct due to the conditional check for resetting only when events are present.   Chris also suggested potentially using a sync primitive after the initial back-off to avoid spinning, using `EnqueueEventsAsync` as the trigger to start processing again.   I'm going to open an issue to track that for an improvement in the next milestone.

Reply:
Ok great, that makes sense! @christothes @jsquire